Mailbox/Person.cs

```diff
@@ -1,4 +1,37 @@
-﻿namespace Mailbox
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace MailRoom
 {
-
-}
\ No newline at end of file
+    public readonly struct Person : IEquatable<Person>
+    {
+        public string FirstName { get; }
+        public string LastName { get; }
+
+        public Person(string firstName, string lastName)
+        {
+            FirstName = firstName ?? throw new ArgumentNullException(nameof(firstName));
+            LastName = lastName ?? throw new ArgumentNullException(nameof(lastName));
+        }
+
+        public override bool Equals(object? obj)
+        {
+            if (obj is null) return false;
+
+            return Equals((Person) obj);

```

Comment:
This will actually throw an exception if a different object instance that is not a `Person` gets passes in.
A better solution would be to use pattern matching.
```suggestion
            return obj is Person person && Equals(person);
```